### PR TITLE
[RFC] Docs update

### PR DESCRIPTION
**Pull request by cipherboy**
#### Why RFC?

The last two headers make this RFC, namely, the contribution guide and placing these in the `gh-pages` branch.

#### Description

Updates to various pieces of documentation and the addition of new documentation.

In particular: 
 - We update the build system documentation to reflect the new `html` target and remove the environment variable introduced by @emaldona for building on Debian as it is no longer used.
 - We update the documentation on using JSS in a downstream project to reflect the new JAXB dependency.
 - We add new documentation on our dependencies, clarifying which are build time / test suite / runtime dependencies.
 - We save the old build instructions to a file called `legacy_building.md` in case anyone wishes to refer to them still. 
- Add documentation on PKCS11 Constants. 
- Update the CI documentation to reflect the new tools
- Add a contributing guide
- Updates to the README

#### Legacy Building

I've moved the build instructions from the README to a separate document. This has been added as a link to the README in case anyone wishes to refer to them. (Since we have JDK9+ support, we're less picky about which version of the JDK we build with so using the `tools/autoenv.sh` should work in most every case). This also lets us update the build documentation without worrying about ordering and corrects a few mistakes in the old documentation.

#### Contributing Guide

I'd like feedback on the contribution guide. It includes links to our IRC / mailing list, and information about opening PRs and issues. For the time being I've added something that says they can file GitHub issues if they don't want a Fedora account, since we _have_ left issues enabled on this repository. If it gets out of hand we can always close one or the other, but I'm guessing outside of our existing contributors, most JSS issues will be filed against Bugzilla anyways.

#### regen-gh-pages

I'd like to place these documents in the GitHub pages branch as well, under a `docs` sub folder so that anyone can refer to them and share them more easily. Thoughts?

### CODE REVIEW

File: docs/dependencies.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Dependencies
+
+## Build-time Dependencies
+
+This project has the following dependencies:
+
+ - [nspr](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/NSPR)
```
Comment (edewata) [inline docs/dependencies.md]: NSPR should be capitalized.

File: docs/dependencies.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ - [nss](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/NSS)
```
Comment (edewata) [inline docs/dependencies.md]: NSS should be capitalized.

File: README.md [hunk 3]
```diff
@@ -32,279 +34,43 @@ To install these dependencies on Fedora, execute the following:
...
-# git checkout -b JSS_4_4_BRANCH
-```
+Building
+--------
 
-### 8. Known Issues
+To build JSS and make a best effort to detect environment variables:
 
-Some of the known issues are:
+    mkdir sandbox/ && cd sandbox
```
[the remark this answers is not in the document]
Comment (cipherboy) [reply README.md]: So if you're building JSS from source and _not_ building an RPM, this is still necessary.

File: docs/using_jss.md [hunk 4]
```diff
@@ -11,3 +13,12 @@ following dependencies are available in your `CLASSPATH`:
    and installed to `/usr/share/java/apache-commons-lang.jar`.
  - `apache-commons-codec.jar` -- provided by the `apache-commons-codec`
    package and installed to `/usr/share/java/apache-commons-codec.jar`.
+ - `jaxb-api.jar` -- provided by the `glassfish-jaxb-api` package
+   and installed to `/usr/share/java/jaxb-api.jar`.
```
[the remark this answers is not in the document]
Comment (cipherboy) [reply docs/using_jss.md]: Updated in the latest commits.

File: docs/building.md [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+To test JSS, run the test suite from the `Makefile` interface:
+
+    make check
+
+### 5. Installing JSS
+
+If JSS already exists on the system, run something similar to the
+following command(s) to back up the old installation:
+
+    sudo mv /usr/lib/java/jss4.jar /usr/lib/java/jss4.jar.orig
```
[the remark this answers is not in the document]
Comment (cipherboy) [reply docs/building.md]: Updated in the latest commits.

File: docs/ci.md [hunk 6]
```diff
@@ -4,21 +4,30 @@ JSS uses [Travis CI](https://travis-ci.org/dogtagpki/jss) for PR gating. We use
...
+Note that this requires either Buildah and Podman or Docker to be installed.
+
 For more extensive documentation, please refer to the
 [Buildah](https://github.com/containers/buildah/blob/master/docs/) and
-[Podman docs](https://github.com/containers/libpod/tree/master/docs).
+[Podman docs](https://github.com/containers/libpod/tree/master/docs) and
+the code in the `tools/run_container.sh` script.
+
+
+## Skipping CI
```
[the remark this answers is not in the document]
Comment (SilleBille) [reply docs/ci.md]: This is a general feature of Travis. If @cipherboy  wants to include it in the doc, I'm not against it!

**Comment by SilleBille** [comment]
This looks neat and detailed! Good one, @cipherboy  :+1:

File: docs/ci.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (edewata) [reply docs/ci.md]: OK, I thought it was only used by Gerrit/Jenkins.

[the remark this answers is not in the document]
Comment (cipherboy) [reply docs/ci.md]: Nah I think a lot of places support this. There's also a `[skip travis]` if we were using other CI builders and only wanted to skip the Travis build.

**Comment by cipherboy** [comment]
Okie dokie, I'm merging then. Thanks @edewata and @SilleBille!

**Pull request merged**